packages/@aws-cdk/aws-ec2/lib/vpc.ts

```diff
@@ -1316,6 +1316,12 @@ export class Subnet extends Resource implements ISubnet {
     return new ImportedSubnet(scope, id, attrs);
   }
 
+  // tslint:disable:no-shadowed-variable
+  public static fromSubnetId(scope: Construct, id: string, subnetId: string): ISubnet {
+    return this.fromSubnetAttributes(scope, id, { subnetId, availabilityZone: '' });
```

Comment:
It *can* be optional in the attrs type, it just  cannot be optional as a class property.

Please make the props type optional.

Reply:
Changed AZ to optional inside SubnetAttributes.